pages/pipelines/hosted_agents/linux.md

```diff
@@ -28,13 +28,106 @@ To configure your Linux instance you can use the [Docker Compose](https://github
+
+### Cache configuration
+
+Cache paths can be [defined in your `pipeline.yml`](/docs/pipelines/defining-steps) file. Defining cache paths for a step will implicitly create a cache volume for the pipeline.
+
+When cache paths are defined, the cache volume is mounted under `/cache` in the agent instance. The agent links subdirectories of the cache volume into the paths specified in the configuration. For example, defining `cache: "node_modules"` in your `pipeline.yml` file will link `./node_modules` to `/cache/bkcache/node_modules` in your agent instance.
+
+Custom caches can be created by specifying a name for the cache, which allows you to use multiple cache volumes in a single pipeline.
+
+When requesting a cache volume, you can specify a size. The cache volume provided will have a minimum available storage equal to the specified size. In the case of a cache hit (most of the time), the actual volume size is: last used volume size + the specified size.
+
+Defining a top-level cache configuration (as opposed to one within a step) sets the defaults cache volume for all steps in the pipeline. Steps can override the top-level configuration by defining their own cache configuration.
+
+```yaml
+cache:
+  paths:
+    - "node_modules"
+  size: "100g"
+
+steps:
+  - command: "yarn run build"
+    cache: ".build"
+
+  - command: "yarn run test"
+    cache:
+      - ".build"
+
+  - command: "rspec"
+    cache:
+      paths:
+        - "vendor/bundle"
+      size: 20g
+      name: "bundle-cache"
+```
+{: codeblock-file="pipeline.yml"}
+
+Required attributes:
+
+<table data-attributes data-attributes-required>
+  <tr>
+    <td><code>paths</code></td>
+    <td>
+      A list of paths to cache. Paths are relative to the working directory of the step.<br>
+      Absolute references can be provided in the cache paths configuration relative to the root of the instance.<br>
+      <em>Example:</em><br>
+      <code>- ".cache"</code><br>
+      <code>- "/tmp/cache"</code><br>
+    </td>
+  </tr>
+</table>
+
+Optional attributes:
+
+<table data-attributes data-attributes-required>
+  <tr>
+    <td><code>name</code></td>
+    <td>
+      A name for the cache. This allows for multiple cache volumes to be used in a single pipeline.<br>
+      <em>Example:</em> <code>"node-modules-cache"</code><br>
+    </td>
+  </tr>
+
+  <tr>
+    <td><code>size</code></td>
+    <td>
+      The size of the cache volume. The default size is 20 gigabytes. Units are in gigabytes, specified as <code>Ng</code>, where <code>N</code> is the size in gigabytes, and <code>g</code> indicates gigabytes.<br>
+      <em>Example:</em> <code>"20g"</code><br>
+    </td>
+  </tr>
+</table>
+
+### Lifecycle
+
+At any point in time, multiple versions of a cache volume may be used by different jobs.
+
+The first request creates the first version of the cache volume, which is used as the parent of subsequent forks until a new parent version is committed.
```

Comment:
What exactly is meant by "fork" in this context? I was thinking of a Git fork, but I can't see how cache volumes are stored in Git repos.

Reply:
Fork of a storage volume 'moment'.